### DISCUSSION TITLE
Include same file multiple times

### DISCUSSION THREAD

**Discussion by michael-projectx** [google-deepmind/mujoco#1063] [General]
I've read the docs & and understand that an XML file can be included at most once, so I'm hoping there's an alternate solution to the problem I'm trying to solve.

I've defined a simulation asset in an XML file that has a number of `<geom>` elements (all attached to a single body). I'd like to include multiple instances of this object within a single scene/mjcf file. It looks like the solution in [this file](https://github.com/google-deepmind/mujoco/blob/main/model/humanoid/22_humanoids.xml) is to define the humanoid object a bunch of times in the same file with unique names for each.

Is there any way to accomplish the task of defining multiple instances of the same "object" in a scene?

I've attached the "object" file that I'd like to include multiple times as an example (I had to use a `.txt` extension so the file would upload, but this would be `plastic_box.xml` in the example below).
[plastic_box.txt](https://github.com/google-deepmind/mujoco/files/12681098/plastic_box.txt)

Ideally I'd be able to use this file something like this:

```
<mujoco model="box_pick_scene">

  <include file="my_robot.mjcf" />

  <worldbody>
    <geom name="floor" class="env_collision" pos="0 0 0" size="0 0 1" type="plane"
      material="groundplane" />
    <geom name="low_table" class="env_collision" pos="10.35 -0.35 0.25" euler="0 0 0.785"
      size="0.1 0.1 0.25" type="box" />

    <body name="tote1" pos="10.35 -0.35 0.3" euler="0 0 2.356">
      <freejoint name="tote1_root_joint" />
      <include file="plastic_box.xml" />
    </body>

    <body name="tote2" pos="10.35 -0.35 0.6" euler="0 0 2.356">
      <freejoint name="tote2_root_joint" />
      <include file="plastic_box.xml" />
    </body>

    <body name="tote3" pos="10.35 -0.35 0.9" euler="0 0 2.356">
      <freejoint name="tote3_root_joint" />
      <include file="plastic_box.xml" />
    </body>

  </worldbody>

</mujoco>
```

**Comment by michael-projectx** [comment]
Maybe I'm going to answer my own question here, but it looks like [pyMJCF](https://github.com/google-deepmind/dm_control/blob/main/dm_control/mjcf/README.md) will provide a solution to my problem, but I'd still like to here if there is an alternate solution to what I'm asking for.

**Reply by yuvaltassa** [reply]
We are planning on adding a <replicate> meta element that will do that (including namespacing), but until that happens PyMJCF is your best bet.
